easybuild/easyblocks/b/boost.py

```diff
@@ -224,57 +223,80 @@ def build_step(self):
+            mpi_bjamoptions = " --with-mpi"
+            self.build_boost_variant(self.bjamoptions + mpi_bjamoptions, self.paracmd)
 
-        if self.cfg['boost_multi_thread']:
-            self.log.info("Building boost with multi threading")
-            self.build_boost_variant(bjamoptions + " threading=multi --layout=tagged", paracmd)
+        self.log.info("Building boost libraries")
+        # build with specified options
+        cmd = "%s ./%s %s %s %s" % (
+            self.cfg['prebuildopts'], self.bjamcmd, self.bjamoptions, self.paracmd, self.cfg['buildopts'])
+        run_cmd(cmd, log_all=True, simple=True)
 
-        # if both boost_mpi and boost_multi_thread are enabled, build boost mpi with multi-thread support
-        if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
-            self.log.info("Building boost_mpi with multi threading")
-            extra_bjamoptions = " --with-mpi threading=multi --layout=tagged"
-            self.build_boost_variant(bjamoptions + extra_bjamoptions, paracmd)
+    def install_step(self):
+        """Install Boost by copying files to install dir."""
 
-        # install remainder of boost libraries
+        # install boost libraries
         self.log.info("Installing boost libraries")
 
         cmd = "%s ./%s %s install %s %s" % (
-            self.cfg['preinstallopts'], self.bjamcmd, bjamoptions, paracmd, self.cfg['installopts'])
+            self.cfg['preinstallopts'], self.bjamcmd, self.bjamoptions, self.paracmd, self.cfg['installopts'])
         run_cmd(cmd, log_all=True, simple=True)
 
-    def install_step(self):
-        """Install Boost by copying files to install dir."""
-
         self.log.info("Copying %s to installation dir %s", self.objdir, self.installdir)
         if self.cfg['only_python_bindings'] and 'Python' in self.cfg['multi_deps'] and self.iter_idx > 0:
             self.log.info("Main installation should already exist, only copying over missing Python libraries.")
-            copy(glob.glob(os.path.join(self.objdir, 'lib', 'libboost_python*')), os.path.join(self.installdir, 'lib'))
+            copy(glob.glob(os.path.join(self.objdir, 'lib', 'libboost_python*')), os.path.join(self.installdir, 'lib'),
+                 symlinks=True)
         else:
-            copy(glob.glob(os.path.join(self.objdir, '*')), self.installdir)
+            copy(glob.glob(os.path.join(self.objdir, '*')), self.installdir, symlinks=True)
+
+        # Link tagged multi threaded libs as the default libs
+        lib_mt_suffix = '-mt'
+        if LooseVersion(self.version) >= LooseVersion("1.69.0"):
+            if get_cpu_architecture() == AARCH64:
+                lib_mt_suffix += '-a64'
+            elif get_cpu_architecture() == POWER:
+                lib_mt_suffix += '-p64'
+            else:
+                lib_mt_suffix += '-x64'
+
+        shlib_ext = get_shared_lib_ext()
+        lib_glob = 'lib*%s.%s.%s' % (lib_mt_suffix, shlib_ext, self.version)
```

Comment:
This replace and glob stuff here gives me headaches. Not sure if all is correct.
Currently we have the following:
- libboost_type_erasure.a
- libboost_type_erasure-mt-x64.a
- libboost_type_erasure-mt-x64.so
- libboost_type_erasure-mt-x64.so.1.74.0
- libboost_type_erasure.so
- libboost_type_erasure.so.1.74.0

So I would say the glob should be `lib*.*` and the replacement simply: "use current name but remove everything from first '-' until first dot".
This removes the need for the lib_mt_suffix and the arch stuff above and makes it way simpler to understand: "Remove the tag"

Reply:
When doing threading=single,multi --layout=tagged you get:
```libboost_type_erasure-mt-x64.a
libboost_type_erasure-mt-x64.so@
libboost_type_erasure-mt-x64.so.1.74.0*
libboost_type_erasure-x64.a
libboost_type_erasure-x64.so@
libboost_type_erasure-x64.so.1.74.0*
```

Reply:
> and nothing else

The "currently" was referring to current EB installations of Boost.

Ah, forgot the single-threaded. Then glob for `lib*-mt*.*` instead. Then 2 comments like "Glob for the mt libs" and "remove the tags from the name" would be enough and the code much easier to understand. Maybe even include those names of yours as examples in a comment

Reply:
yeah, testing...

Reply:
Better now I hope...